code/modules/shuttle/shuttle.dm

```diff
@@ -506,41 +598,46 @@
 				if(A.locked)
 					A.unlock()
 
-/obj/docking_port/mobile/proc/roadkill(list/L, dir, x, y)
-	for(var/turf/T in L)
-		for(var/atom/movable/AM in T)
+/obj/docking_port/mobile/proc/roadkill(list/L0, list/L1, dir)
+	var/list/hurt_mobs = list()
+	for(var/i in 1 to L0.len)
+		var/turf/T0 = L0[i]
+		var/turf/T1 = L1[i]
+		if(!T0 || !T1)
+			continue
+		//if(T0.type == T0.baseturf)//wot?
+		//	continue
+		// The corresponding tile will not be changed, so no roadkill
+
+		for(var/atom/movable/AM in T1)
 			if(ismob(AM))
-				if(istype(AM, /mob/living))
+				if(isliving(AM) && !(AM in hurt_mobs))
+					hurt_mobs |= AM
 					var/mob/living/M = AM
-					M.Paralyse(10)
-					M.take_organ_damage(80)
-					M.anchored = 0
-				else
+					if(M.buckled)
+						M.buckled.unbuckle_mob(M, 1)
+					if(M.pulledby)
+						M.pulledby.stop_pulling()
+					M.stop_pulling()
+					M.visible_message("<span class='warning'>[M] is hit by \
+							a hyperspace ripple[M.anchored ? "":" and is thrown clear"]!</span>",
+							"<span class='userdanger'>You feel an immense \
+							crushing pressure as the space around you ripples.</span>")
+					if(M.anchored)
+						M.gib()
+					else
+						step(M, dir)
```

Comment:
I'm pretty sure the placement of this will mean that mobs that are in the center of a docking shuttle will only be moved once and paralyzed once, therefore most likely ending up inside of the shuttle, instead of being thrown completely clear of it.